temporal-sdk/src/main/java/io/temporal/internal/worker/ShutdownManager.java

```diff
@@ -0,0 +1,161 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.worker;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.concurrent.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ShutdownManager implements Closeable {
+  private static final Logger log = LoggerFactory.getLogger(ShutdownManager.class);
+
+  private final ScheduledExecutorService scheduledExecutorService =
```

Comment:
Yeah, but it's the only way how I see it can be implemented. I need a thread to wait on completion and fill the CompletableFuture. 
It's fine though I think. I create this instance ONLY during shutdown for the time of shutdown.
And... it's not the first "managing" thread we have.